cogs/respects/respects.py

```diff
@@ -161,87 +173,107 @@ async def setfTime(self, ctx, seconds: int):
             seconds,
         )
 
-    async def checkLastRespect(self, ctx):
+    async def checkLastRespect(self, ctx: Context):
         """Check to see if respects have been paid already.
 
         This method only checks the time portion, previous messages
         and the last respect.
 
-        This method returns False if:
-        - No respects have been paid in this channel before, or
+        Returns:
+        --------
+        This method returns `False` if:
+        - No respects have been paid in the channel before, or
+        - The last respect could not be retrieved, or
         - The current respect is paid to a message that is different from
           the message to which the last respect was paid, or
         - The time exceeds the threshold AND the last respect in the channel was behind
           more than a certain number of messages.
 
-        Otherwise, this method returns True.
+        Otherwise, this method returns `True`.
         """
-        chData = await self.config.channel(ctx.channel).all()
-        guildData = await self.config.guild(ctx.guild).all()
 
-        if not chData[KEY_MSG]:
+        chConfig = self.config.channel(ctx.channel)
+        guildConfig = self.config.guild(ctx.guild)
+
+        oldRespectMsgId = None
+        try:
+            oldRespectMsgId = await chConfig.get_attr(KEY_MSG)()
+        except AttributeError:
+            # the attribute has not been registered
+            # so leave it as is
+            pass
```

Comment:
Do we need the except block? We shouldn't hit this because we registered KEY_MSG when we initialized the cog, on line 44.

Reply:
Oh true. `Red` should fallback to the default values registered. Will commit some changes soon.